e2e/various-suite/frontend-sandbox-datasource.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { random } from 'lodash';
+
+import { e2e } from '../utils';
+
+const DATASOURCE_ID = 'sandbox-test-datasource';
+let DATASOURCE_CONNECTION_ID = '';
+const DATASOURCE_TYPED_NAME = 'SandboxDatasourceInstance';
+
+describe('Datasource sandbox', () => {
+  before(() => {
+    e2e.flows.login(Cypress.env('USERNAME'), Cypress.env('PASSWORD'), true);
+
+    e2e.pages.AddDataSource.visit();
+    e2e.pages.AddDataSource.dataSourcePluginsV2('Sandbox datasource test plugin')
+      .scrollIntoView()
+      .should('be.visible') // prevents flakiness
+      .click();
+    e2e.pages.DataSource.name().clear();
+    e2e.pages.DataSource.name().type(DATASOURCE_TYPED_NAME);
+    e2e.pages.DataSource.saveAndTest().click();
+    cy.url().then((url) => {
+      const split = url.split('/');
+      DATASOURCE_CONNECTION_ID = split[split.length - 1];
+    });
+  });
+  beforeEach(() => {
+    e2e.flows.login(Cypress.env('USERNAME'), Cypress.env('PASSWORD'), true);
+  });
+
+  describe('Config Editor', () => {
+    describe('Sandbox disabled', () => {
+      beforeEach(() => {
+        cy.window().then((win) => {
+          win.localStorage.setItem('grafana.featureToggles', 'pluginsFrontendSandbox=0');
+        });
+      });
+      it('Should not render a sandbox wrapper around the datasource config editor', () => {
+        cy.visit('connections/datasources/edit/' + DATASOURCE_CONNECTION_ID);
```

Comment:
Use `e2e.pages.EditDataSource.visit(DATASOURCE_CONNECTION_ID)` instead?

Reply:
Amazing! I don't even know how to discover these. thanks for pointing it up.